torch/_library/triton.py

```diff
@@ -90,7 +93,18 @@ def triton_op(
     """
 
     def dec(fn: Callable) -> Callable:
-        result = custom_op(name, fn, mutates_args=mutates_args)
+        def backend_fn(*args, **kwargs):  # type: ignore[no-untyped-def]
+            # Optimization: we're passing regular Tensors into the triton kernel, so
+            # no need to go through HOP dispatch
+            with set_capture_triton_enabled(False):
+                return fn(*args, **kwargs)
+
+        result = custom_op(
+            name,
+            backend_fn,
```

Comment:
what does the backend_fn do?

Reply:
We register backend_fn to the CompositeExplicitAutograd key. Those are guaranteed to run after tracing / subclasses/ etc are done, so it's safe to dispatch directly to the triton kernel